src/main/java/life/qbic/subscriptions/encryption/Decrypter.java

```diff
@@ -0,0 +1,25 @@
+package life.qbic.subscriptions.encryption;
+
+import java.util.function.Function;
+
+/**
+ * Decrypts a message of type {@link I} to a message of type {@link O}
+ *
+ * @param <I> encoded input type
+ * @param <O> decoded output type
+ */
+public interface Decrypter<I, O> extends Function<I, O> {
+
+  /**
+   * Decrypts a message of type {@link I} to a message of type {@link O}
+   *
+   * @param message encrypted message
+   * @return original value
+   */
+  O decrypt(I message);
```

Comment:
Shouldn't this be a private method? Then we would only expose the apply method --> functional interface

Reply:
@KochTobi she has a point, can you comment on her question?

Reply:
the decrypt replaces the apply method